### PR TITLE
highlight the selected user icon & mobile-size

### PR DESCRIPTION
**Pull request by tyxxxx**
Hi All,

1. highlighting the selected user icon when filtering by user so that users can see when there is a search happening
2. Don’t allow typing non-numerical text into price filters
3. As Barney said last week: when the interface is the mobile-size, users need to click user list to load user
4. Adding a scroll bar to the user list(When the screen is half the size, you can see the scroll bar)

屏幕快照 2021-08-05 上午4 51 32

屏幕快照 2021-08-05 上午4 51 41

### CODE REVIEW

File: vanhouse/src/components/UserList.js [hunk 1]
```diff
@@ -45,12 +60,42 @@ function UserList({ setQuery, setUserId }) {
...
 
     return (
-        <div className="user-box">
-            <h2>User List</h2>
-            <Container fluid>{list}</Container>
+        <div>
 
-            <div className="cancel-btn">
-                <button type="button" onClick={Cancel}>Cancel</button>
+            <div className="user-box user_scroll_div d-none d-md-block">
```
Comment (BarneyShi) [inline vanhouse/src/components/UserList.js:65]: @tyxxxx In this div there are some `class` attributes, in react we should use className.

File: vanhouse/src/components/UserList.js [hunk 2]
```diff
@@ -45,12 +60,42 @@ function UserList({ setQuery, setUserId }) {
...
+                    <div class="card">
+                        <div class="card-header" id="headingOne">
+                        <h2 class="mb-0"  data-toggle="collapse" data-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
+                            User List
+                        </h2>
+                        </div>
+
+                        <div id="collapseOne" class="collapse" aria-labelledby="headingOne" data-parent="#user-list">
+                            <div class="card-body">
+                            {list}
```
Comment (BarneyShi) [inline vanhouse/src/components/UserList.js:94]: @tyxxxx user list won't show after clicking on it, might be a bug.

File: vanhouse/src/components/UserList.js [hunk 3]
```diff
@@ -32,11 +49,9 @@ function UserList({ setQuery, setUserId }) {
                 const newName = [];
                 console.log(res);
                 const cardList = res.map((item, index) => (
-                    <div className="user-item" key={item} >
+                    <div className="user-item" key={item}  onClick={(e) => filterUserPost(e, item)}>
```
Comment (BarneyShi) [inline vanhouse/src/components/UserList.js:52]: @tyxxxx  Let's use `item._id` as the key, otherwise it shows errors in the console.

**Review by BarneyShi — COMMENTED** [review]
It looks good but needs some tweaks.

**Comment by NaithanB** [comment]
Hi Yuxin,
Thanks! The user list looks nice! The city selector looks way nicer on Safari now too. 

Just a couple things that I think could be changed:

- When I have a mobile sized view, the user list cuts off and only shows the header. I can’t scroll down to see all the users. (See the image. I'm scrolled down as far as possible)

vhSc1

- For the UserList, I think it might be good to make the margin for the inner div extend closer to the edge of the scroll window (or possibly all the way to the edge if it looks good). That way when the browser window is more narrow, there’s more space to display the user names and the text won’t need to wrap quite as quickly.
- The price input fields could be just slightly wider so that 4 digits are visible when entering the price.
-  Just as an idea - it might be pretty nice if clicking inside the userList component but not on a particular user would deselect any selected users like the user panel cancel button used to do.

**Comment by NaithanB** [comment]
When I'm in mobile sized view, I'm not sure it's obvious that that clicking on the user list will expand it. Is there a way to show visually that it's clickable? Maybe a dropdown arrow icon or something? It is a nice addition though!

**Comment by tyxxxx** [comment]
@NaithanB Sure! can you check the message on slack？

**Review by BarneyShi — APPROVED** [review]

**Review by parksj23 — APPROVED** [review]

**Comment by parksj23** [comment]
@tyxxxx I have the same suggestions as Naithan so I've approved the changes since you are already aware of them. In particular, if we could click on the user again to deselect (or something like that), and add something to the user list in mobile view to make it clear that it will expand if clicked. Otherwise it looks great!

**Comment by tyxxxx** [comment]
Hi All,

Now
1. the price input fields are wider 
2. For user list, the margin for the inner div extend closer to the edge 
3. I add an icon for user list, so it's obvious that that clicking on the user list will expand it

I don't think the function of "click inside the userlist component but not on a specific user" will be done for the time being. I don't know how to do it at present haha. So keep it like this. 
You can test it. I think it is ready for merge @NaithanB @parksj23 @BarneyShi 

I will open a new pull request, which is the function of switching Chinese. I have done most of it, but the account interface has not started. And there are other small problems. I can't open the account interface here. It is shows "still under construction". I think I need Barney's help for doing this part

**Review by NaithanB — APPROVED** [review]
Great! Looks nice.

**Pull request merged**